lib/spack/spack/platforms/cray.py

```diff
@@ -106,7 +114,7 @@ def setup_platform_environment(cls, pkg, env):
 
     @classmethod
     def detect(cls):
-        return os.environ.get('CRAYPE_VERSION') is not None
+        return 'CRAYPE_VERSION' in os.environ or os.path.exists('/opt/cray/pe')
```

Comment:
This change seems to be fine, in the sense that it returns `True` even after a `module purge`. I am wondering if this can be further simplified to:
```suggestion
        return os.path.exists('/opt/cray/pe')
```
but I am not sure of it as I have currently access only to Cori.

Reply:
On Cori after a `module purge` the only `CRAY_*` variable available is:
```console
culpo@cori10:~/github/spack> echo $CRAY_SITE_LIST_DIR
/etc/opt/cray/pe/modules
```
so maybe this might also be a good candidate to check if platform is `cray`? To be clear, I'm not suggesting a change here, just noting that a variable stays there after we pruned all the modules.

Reply:
That variable is not set on the machine I've been testing on.

Reply:
Should the check really be `'/opt/cray' in os.environ.get('MODULEPATH', '')`? I think the fundamental quality of a cray platform is the availability of the PE through modules.